WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -1,47 +1,44 @@
 import Foundation
 
 
-
-// MARK: - ManagedObject Base Class
+// MARK: - NSManagedObject Default entityName Helper
 //
-protocol ManagedObject {
-    static var entityName: String { get }
+extension NSManagedObject {
+    class var entityName: String {
+        return classNameWithoutNamespaces()
+    }
 }
 
 
-// MARK: - Core Data Helpers
+// MARK: - NSManagedObjectContext Helpers!
 //
-struct CoreDataHelper<T> where T: NSManagedObject, T: ManagedObject {
-    /// CoreData ManagedObjectContext
-    ///
-    let context: NSManagedObjectContext
-
+extension NSManagedObjectContext {
 
     /// Returns a new FetchRequest for the associated Entity
     ///
-    func newFetchRequest() -> NSFetchRequest<NSFetchRequestResult> {
-        return NSFetchRequest(entityName: T.entityName)
+    func newFetchRequest<T: NSManagedObject>(forType type: T.Type) -> NSFetchRequest<NSFetchRequestResult> {
```

Comment:
Good call! This was initially implemented when we had iOS 9 support on the table, and definitely forgot about the iOS 10 helper. Replacing, thanks for spotting this!!